drivers/sensor/fxas21002/fxas21002.c

```diff
@@ -18,6 +18,131 @@ static const uint32_t sample_period[] = {
 	1250, 2500, 5000, 10000, 20000, 40000, 80000, 80000
 };
 
+#if DT_ANY_INST_ON_BUS_STATUS_OKAY(spi)
+#define DIR_READ(a)		((a) | (1 << 7))
+#define DIR_WRITE(a)		((a) & 0x7f)
+
+static int fxas21002_transceive(const struct device *dev,
+				void *data, size_t length)
+{
+	const struct fxas21002_config *cfg = dev->config;
+	const struct spi_buf buf = { .buf = data, .len = length };
+	const struct spi_buf_set s = { .buffers = &buf, .count = 1 };
+
+	return spi_transceive_dt(&cfg->bus_cfg.spi, &s, &s);
+}
+
+int fxas21002_read_spi(const struct device *dev,
+		       uint8_t reg,
+		       void *data,
+		       size_t length)
+{
+	const struct fxas21002_config *cfg = dev->config;
+
+	/* Reads must clock out a dummy byte after sending the address. */
+	uint8_t reg_buf[2] = { DIR_READ(reg), 0 };
+	const struct spi_buf buf[2] = {
+		{ .buf = reg_buf, .len = 3 },
+		{ .buf = data, .len = length }
+	};
+	const struct spi_buf_set tx = { .buffers = buf, .count = 1 };
+	const struct spi_buf_set rx = { .buffers = buf, .count = 2 };
+
+	return spi_transceive_dt(&cfg->bus_cfg.spi, &tx, &rx);
+}
+
+int fxas21002_byte_read_spi(const struct device *dev,
+			    uint8_t reg,
+			    uint8_t *byte)
+{
+	/* Reads must clock out a dummy byte after sending the address. */
+	uint8_t data[] = { DIR_READ(reg), 0};
+	int ret;
+
+	ret = fxas21002_transceive(dev, data, sizeof(data));
+
+	*byte = data[1];
+
+	return ret;
+}
+
+int fxas21002_byte_write_spi(const struct device *dev,
+			     uint8_t reg,
+			     uint8_t byte)
+{
+	uint8_t data[] = { DIR_WRITE(reg), byte };
+
+	return fxas21002_transceive(dev, data, sizeof(data));
+}
+
+int fxas21002_reg_field_update_spi(const struct device *dev,
+				   uint8_t reg,
+				   uint8_t mask,
+				   uint8_t val)
+{
+	uint8_t old_val;
+
+	if (fxas21002_byte_read_spi(dev, reg, &old_val) < 0) {
+		return -EIO;
```

Comment:
Please propagate error codes, don't do conversions as you're loosing information.